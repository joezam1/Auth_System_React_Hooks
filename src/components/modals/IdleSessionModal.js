import React, { useEffect, useState } from "react";
import SessionConfig from '../../../configuration/authentication/SessionConfig.js';
import LocalStorageService from "../../services/localStorage/LocalStorageService";



//Test: DONE
export default function IdleSessionModal(){


    let _countdownValue = 100;
    const [countdownValue, setCountdown] = useState();

    useEffect(()=>{
        console.log('idleSessionModal-useEffect-triggered-OK');
        let _intervalId = setInterval(() => {
            _countdownValue = LocalStorageService.getItemFromLocalStorage(SessionConfig.IDLE_SESSION_COUNTDOWN_VALUE);
            console.log('useEffect-setInterval-countdownValue', _countdownValue);
            setCountdown(_countdownValue);
            if(_countdownValue <= 0){
                clearInterval(_intervalId);
                console.log('clearInterval-DONE-_intervalId', _intervalId);
                console.log('clearInterval-DONE-_countdownValue', _countdownValue);
            }
        }, SessionConfig.ONE_SECOND_IN_MILLISECONDS);
        console.log('idleSessionModal-useEffect-_intervalId', _intervalId)
    }, []);

    return(<div className="idle-session-modal-section">
        <div className="modal-container">
            <div className="modal-notification-box">
                <div className="modal-notification-text">
                    You will be Auto looged out in <b>{countdownValue ? countdownValue + ' seconds.': '' }</b>
                    <br/>
                    Please generate activity in your browser to maintain your session.
                </div>
            </div>
        </div>
    </div>);
}
