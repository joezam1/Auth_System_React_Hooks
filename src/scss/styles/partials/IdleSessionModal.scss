
.modal-container{
    background-color: transparent;
    border: 0px solid blue;
    position: absolute;
    text-align: center;
    width: 100vw;
    height:100vh;
    z-index:500;
}


.logout-warning-box{
    border: 1px solid silver;
    border-radius:5px;
    //box-shadow: rgb(38, 57, 77) 0px 20px 30px -10px;
    box-shadow: rgba(0, 0, 0, 0.3) 0px 19px 38px, rgba(0, 0, 0, 0.22) 0px 15px 12px;
    //box-shadow: rgba(0, 0, 0, 0.25) 0px 54px 55px, rgba(0, 0, 0, 0.12) 0px -12px 30px, rgba(0, 0, 0, 0.12) 0px 4px 6px, rgba(0, 0, 0, 0.17) 0px 12px 13px, rgba(0, 0, 0, 0.09) 0px -3px 5px;
    background-color: white;
    display: inline-block;
    height: 50vh;
    max-height:150px;
    position:relative;
    top:30vh;
}

.logout-warning-text{
    margin: 50px;
    text-align: left;
}
