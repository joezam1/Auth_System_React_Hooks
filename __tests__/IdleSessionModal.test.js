import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';

import IdleSessionModal from "../src/components/modals/IdleSessionModal.js";


describe('File: IdleSessionModal.js', function(){
    afterEach(cleanup);
    afterAll(()=>{jest.resetAllMocks(); });

    describe('Component: IdleSessionModal', function(){

        test('It can take a snapshot', function(){
            let container = render( <MemoryRouter> <IdleSessionModal /> </MemoryRouter>);
            expect(container.asFragment(<IdleSessionModal/> )).toMatchSnapshot();
        });
    });
});
