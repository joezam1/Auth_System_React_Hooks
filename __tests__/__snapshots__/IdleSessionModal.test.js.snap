// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File: IdleSessionModal.js Component: IdleSessionModal It can take a snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="idle-session-modal-section"
  >
    <div
      class="modal-container"
    >
      <div
        class="modal-notification-box"
      >
        <div
          class="modal-notification-text"
        >
          You will be Auto looged out in 
          <b />
          <br />
          Please generate activity in your browser to maintain your session.
        </div>
      </div>
    </div>
  </div>
   
</DocumentFragment>
`;
